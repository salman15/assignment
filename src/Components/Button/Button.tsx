import React from "react";

interface ButtonProps {
  children: React.ReactNode;
  action: Function;
}

const Button: React.FC<ButtonProps> = ({ children, action }) => {
  return <button onClick={() => action()}>{children}</button>;
};

export default Button;
