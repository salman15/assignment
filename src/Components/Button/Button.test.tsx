import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Button from "./Button";

describe("Test the Button component", () => {
  it("Test if Button component renders with Text", () => {
    // Act
    const { container, getByText } = render(<Button>I'm a button</Button>);

    // Assert
    const textValue = getByText(/I'm a button/i);
    expect(container).toMatchSnapshot();
    expect(textValue).toBeInTheDocument();
  });

  it("Test if Button component element changes to an anchor", () => {
    // Act
    const { container } = render(<Button link="#">I'm a button</Button>);

    // Assert
    const element = container.querySelector("a");
    expect(container).toMatchSnapshot();
    expect(element).toHaveTextContent("I'm a button");
  });
});
