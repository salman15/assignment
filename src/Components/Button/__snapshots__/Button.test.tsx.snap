// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Button component Test if Button component element changes to an anchor 1`] = `
<div>
  <a
    class="sc-bdfBwQ sc-gsTCUz iDdfZq bhjXvn"
    href="#"
  >
    I'm a button
  </a>
</div>
`;

exports[`Test the Button component Test if Button component renders with Text 1`] = `
<div>
  <button
    class="sc-bdfBwQ sc-gsTCUz iDdfZq bhjXvn"
    href=""
  >
    I'm a button
  </button>
</div>
`;
